docker/package/tezos_baking_wizard.py

```diff
@@ -0,0 +1,489 @@
+        intrange = list(map(str, range(1, len(options) + 1)))
+        if input not in intrange and input not in options:
+            raise ValueError(
+                "Please choose one of the provided values or use their respective numbers: "
+                + ", ".join(options)
+            )
+        try:
+            opt = int(input) - 1
+        except:
+            return options[input]
+        else:
+            return list(options.keys())[opt]
+
+    return _validator
+
+
+def filepath_validator(input):
+    if input and not os.path.isfile(input):
+        raise ValueError("Please input a valid file path.")
+    return input
+
+
+def required_field_validator(input):
+    if not input.strip():
+        raise ValueError("Please provide this required option.")
+    return input
+
+
+class Validator:
+    def __init__(self, validator):
+        self.validator = validator
+
+    def validate(self, input):
+        if self.validator is not None:
+            if isinstance(self.validator, list):
+                for v in self.validator:
+                    input = v(input)
+                return input
+            else:
+                return self.validator(input)
+
+
+# Wizard CLI utility
+
+
+def proc_call(cmd):
+    return subprocess.check_call(shlex.split(cmd))
+
+
+def fetch_snapshot(url):
+    print("Downloading the snapshot from ", url)
+    filename = "/tmp/tezos_node.snapshot"
+    urllib.request.urlretrieve(url, filename, progressbar_hook)
+    print()
+    return filename
+
+
+def progressbar_hook(chunk_number, chunk_size, total_size):
+    done = chunk_number * chunk_size
+    percent = min(int(done * 100 / total_size), 100)
+    print("Progress: ", percent, "%, ", int(done / (1024 * 1024)), "MB", end="\r")
+
+
+def pprint_table(options):
+    result = "\n"
+    i = 1
+    str_format = "{:1}. {:<20}  {}\n"
+    for o in options:
+        description = textwrap.indent(textwrap.fill(options[o], 60), " " * 25).lstrip()
+        result += str_format.format(i, o, description)
+        i += 1
+
+    return result
+
+
+def color(input, colorcode):
+    return colorcode + input + "\x1b[0m"
+
+
+class Step:
+    def __init__(
+        self,
+        id: str,
+        prompt: str,
+        help: str,
+        default: str = "1",
+        options=None,
+        validator=None,
+        reqs=lambda _vs: True,
+    ):
+        self.id = id
+        self.prompt = prompt
+        self.help = help
+        self.default = default
+        self.options = options
+        self.validator = validator
+        self.reqs = reqs
+
+
+def run_wizard(networks):
+    print("Tezos Baking Wizard")
+    print()
+    print(
+        "Welcome to a wizard helping you to set up the infrastructure for interacting with the Tezos blockchain."
+    )
+    print(
+        "In order to run a baking instance, you'll need the following Tezos binaries:",
+        "tezos-client, tezos-node, tezos-baker-<proto>, tezos-endorser-<proto>.",
+    )
```

Comment:
Perhaps we should mention that if one has installed `tezos-baking` they are all already present?